op-node,batcher,proposer,challenger: Add env var flag tests

trianglesphere opened the pull request:
**Description**

Add a check to ensure that the env var name matches the flag name. Also adds flag tests to op-batcher and op-proposer

ajsutton commented:
Yeah I like this check even if there are a lot of exceptions it will mean new flags are sane at least.

I also wonder if we can/should introduce a helper utility to automatically add env vars to options. Not for this PR and it would require some playing around with to get a good api (which may not be possible) but it would be cool if we could pass a simple definition of the cli arg to a function that then generates a correct cli flag declaration for it. It could automatically calculate and add the env var and potentially some other things we often get wrong. Vague idea currently but I hate how error prone and manual CLI flag declarations are right now...

trianglesphere commented:
@ajsutton I'd like to be able to auto-generate env vars from the flag names, but right now there would be lots of exceptions.

We've got a couple different ways of defining flags. I wonder if we create our own & then have helper functions turn it into what urfave expects

ajsutton commented:
> We've got a couple different ways of defining flags. I wonder if we create our own & then have helper functions turn it into what urfave expects

Yeah that's kind of where my thought process winds up. We basically have a wrapper around urfave that makes it easy and intuitive to define flags that work well and just call a `ToUrfave()` kind of method to get the resulting flag definition.

In terms of exceptions we could either support overriding defaults or just not use the helper in places where we've done something weird that we're stuck with.  We'd want to be able to mix and match anyway so we can incrementally convert to the new approach anyway rather than having to change every CLI option all at once.

ajsutton commented:
Not a small project mind you, but its exactly the kind of refactoring I like to solve a whole class of problem by making it harder to do the wrong thing than to do the right thing.

trianglesphere committed:
op-node,batcher,proposer,challenger: Add env var flag tests

This creates a standardized conversion between env vars and flag names and asserts
that most existing flags and all new flags satisfy the format. Existing flags that
don't match are skipped. This also adds general flag tests for op-batcher and
op-proposer.

ajsutton commented on op-batcher/flags/flags_test.go:
```diff
@@ -0,0 +1,99 @@
+func TestUniqueFlags(t *testing.T) {
+	seenCLI := make(map[string]struct{})
+	for _, flag := range Flags {
+		name := flag.Names()[0]
```
nit: Shouldn't we check that all names are unique, not just the primary names?

trianglesphere replied:
good idea. Typically we only have a single flag name, but the API behind urfave forces you to have multiple (b/c of aliases)

ajsutton commented on op-batcher/flags/flags_test.go:
```diff
@@ -0,0 +1,99 @@
+		flag := flag
+		flagName := flag.Names()[0]
+
+		skippedFlags := []string{
+			txmgr.FeeLimitMultiplierFlagName,
+			txmgr.TxSendTimeoutFlagName,
+			txmgr.TxNotInMempoolTimeoutFlagName,
+		}
```
Q: Why not just add an env far for these?

trianglesphere replied:
They have env var (see the above test), but don't match the flag name -> env var conversion

ajsutton reviewed (approved):
LGTM.  If it's easy to just add the missing env vars I'd be tempted to do that instead of adding the exceptions for them.  But that could be a follow up PR.  At minimum we should ensure we have an open issue for it.
